File: app/src/main/java/shaishav/com/bebetter/receiver/BootReceiver.java
package shaishav.com.bebetter.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.widget.Toast;

import shaishav.com.bebetter.service.UsageService;

/**
 * Created by shaishavgandhi05 on 11/5/16.
 */

public class BootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Toast.makeText(context, "Started", Toast.LENGTH_SHORT).show();
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(new Intent(context, UsageService.class));
        } else {
            context.startService(new Intent(context, UsageService.class));
        }
    }
}

File: app/src/main/java/shaishav/com/bebetter/receiver/BootReceiver.kt
package shaishav.com.bebetter.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.widget.Toast
import shaishav.com.bebetter.service.UsageService
import shaishav.com.bebetter.service.WorkflowService
import shaishav.com.bebetter.utils.BBApplication
import shaishav.com.bebetter.workflow.UsageWorkflow
import java.util.*
import javax.inject.Inject

/**
 * Created by shaishav.gandhi on 2/23/18.
 */
class BootReceiver : BroadcastReceiver() {


  override fun onReceive(context: Context, intent: Intent) {
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
      context.startForegroundService(Intent(context, UsageService::class.java))
    } else {
      context.startService(Intent(context, UsageService::class.java))
    }

    val serviceIntent = Intent(context, WorkflowService::class.java)
    serviceIntent.action = WorkflowService.ACTION_ON
    context.startService(serviceIntent)
  }
}